src/twisted/internet/abstract.py

```diff
@@ -18,6 +18,11 @@
 from twisted.internet import interfaces, main
 
 
+def _dataMustBeBytes(obj):
+    if not isinstance(obj, bytes):  # no, really, I mean it
+        raise TypeError("Data must be bytes")
```

Comment:
Abstracted out so that he error message raised is consistent.

This used to check for string (in the various places this code existed), which caused Mypy to complain that the `raise` line is unreachable, since the tested object was typed as `bytes`.

Changing it to test for `bytes` instead got Mypy not to complain (presumably because type-checking-and-raising is common), and it has the advantage of raising for other invalid types, not just `str`; if we expect `bytes`, that's what we should be testing for.